### DISCUSSION TITLE
Doubt Regarding - Detection and Recognition model For PaddleOCR

### DISCUSSION THREAD

**Discussion by kishore-work-hard** [PaddlePaddle/PaddleOCR#13099] [Q&A]
I have a dataset of cropped number plates - quantity - 10,000 images. 
please note that these number plates are cropped using YOLO v8 model. 

Should I train Both text detection and text recognition  model for paddle OCR? 
or Should I train only text recognition (since plate detection is done by YOLO)

I trained the recognition model with 10k number plate images. but the results are worse than the default paddle model. 
please enlighten me on what went wrong.

**Comment by SWHL** [comment]
You can directly only use text recognition. .

Please provider more info about your trained recognition model

**Reply by kishore-work-hard** [reply]
> You can directly only use text recognition. .
> 
> Please provider more info about your trained recognition model

![a (1)](https://github.com/PaddlePaddle/PaddleOCR/assets/93419928/a0065c23-1119-4b9d-9282-b58ed2860516)
![a (2)](https://github.com/PaddlePaddle/PaddleOCR/assets/93419928/451c9341-50ae-4c87-9624-ae7f85308919)
![a (3)](https://github.com/PaddlePaddle/PaddleOCR/assets/93419928/9b4ef325-5ed2-4ed8-87b1-dcf48fcb0738)
![a (4)](https://github.com/PaddlePaddle/PaddleOCR/assets/93419928/6f5362a1-20de-4534-8676-123eda0c45ae)
![a (5)](https://github.com/PaddlePaddle/PaddleOCR/assets/93419928/282e998c-8cfb-489e-bbeb-85084ff3b039)
![a (6)](https://github.com/PaddlePaddle/PaddleOCR/assets/93419928/00e9994f-dbb5-4f3c-8b02-f816f809cc0f)
![a (7)](https://github.com/PaddlePaddle/PaddleOCR/assets/93419928/4206fccc-a408-439e-a964-e4c3a17f6e6f)
![b](https://github.com/PaddlePaddle/PaddleOCR/assets/93419928/737bbf92-d561-459e-b8ab-895f70d144bb)
![c](https://github.com/PaddlePaddle/PaddleOCR/assets/93419928/8db0b5c6-3f65-4304-b90b-7f799fb33b1c)


provided above are the sample cropped number plates trained for rec model. 

used dictionary - 
[dict.txt](https://github.com/user-attachments/files/15890159/dict.txt)

yml file used for training ocr
[en_PP-OCRv3_rec.yml.txt](https://github.com/user-attachments/files/15890190/en_PP-OCRv3_rec.yml.txt)
but i used 1000 epoch instead of 500.

results for OCR are usually empty string. sometimes i get a character or two out of 10 characters. default paddle can detect almost all character even if the recognition is wrong.

HOW MANY NUMBER PLATE IMAGES ARE NEEDED IN TRAIN FOLDER. MIN EXPECTED COUNT TO GET SATISFACTORY RESULTS?

**Comment by UserWangZz** [comment]
I see that your license plate data has a large projection transformation. You can try to use the det model to detect and correct the image, and then crop to identify the image.

**Reply by kishore-work-hard** [reply]
> I see that your license plate data has a large projection transformation. You can try to use the **det** model to detect and correct the image, and then crop to identify the image.

what do you mean by projection transformation? number plates are in a different angle (it's not horizontal) ? 

can you tell me step by step what to do. 


is this your suggested steps -
1 - yolo v8 crops number plates from cars and bikes. 
2 - draw bounding boxes for all texts inside the number plates (one bbox for each line of text)
3 - use the cropped number plates and it's annotations as dataset for paddle **det** model
4 - train the paddle text **det** model
5 - use the cropped number plates as dataset for paddle **rec** model
6 - train the paddle rec model
7 - use yolo as plate det model, enable paddle as rec model, enable paddle as text det model 

?


or you are asking me to do something with the yolo plate detected plates before sending it to paddle recognition training? @UserWangZz

**Reply by pcycccccc** [reply]
@kishore-work-hard 
My understanding is that you are using crop_images for training, where the text is not horizontal and has some projection transformation relative to the image. Directly training with such data might not yield good results. YOLOv8 only outlines the position of the target and does not correct the transformation of the text on your images. Personally, I feel that YOLOv8 object detection is not suitable for your dataset scenario.

**Reply by pcycccccc** [reply]
@kishore-work-hard My suggestion is that you can use PPOCRLabel to annotate your uncropped images (the whole images). Prepare the dataset according to the user guide, which will automatically generate new crop_images based on your images and labels. Train the rec model with these data. However, this is just my suggestion, as I have only recently started learning about OCR models.

**Reply by kishore-work-hard** [reply]
> @kishore-work-hard My suggestion is that you can use PPOCRLabel to annotate your uncropped images (the whole images). Prepare the dataset according to the user guide, which will automatically generate new crop_images based on your images and labels. Train the rec model with these data. However, this is just my suggestion, as I have only recently started learning about OCR models.

I will try it out @pcycccccc . 
@UserWangZz  what's your opinion ?

**Reply by kishore-work-hard** [reply]
@UserWangZz ?

**Reply by UserWangZz** [reply]
For the workload of annotating the dataset, you need to have an evaluation. You can also try using a text detection model on existing images to see if it can achieve a better text detection effect. If the text detection effect does not meet expectations, it is recommended that you try data annotation.
Sorry, busy with work for me, so I couldn't reply to your message in a timely manner.

**Reply by UserWangZz** [reply]
Referring to the code in predict_sys.py, after passing through the text detection model, certain corrections will be made to the detection area to obtain the cropped image.
Refer to the code block below：
https://github.com/PaddlePaddle/PaddleOCR/blob/main/tools/infer/utility.py#L636-L677https://github.com/PaddlePaddle/PaddleOCR/blob/433677182f108c1be413ee8a92815bc13b205737/tools/infer/utility.py#L636-L677

**Reply by kishore-work-hard** [reply]
@UserWangZz 

I have some questions that's needs some answers 

1-
Is it ok to train rec model using cropped number plate images ? 

2-
Or is it necessary to use the full sized image (without cropping licence plate) and use ppocr label to annotate the texts within licence plate ? And then use it for training? Is it necessary?

3-
For the default paddle ocr model (English) how many images were used to train ?? 

4-
For my specific scenario, how many licence plate images should I train for attaining the best possible results ? (In ur opinion) Please give a number 

5-
Yolo object detection only uses rectangular bounding boxes. 
But when using paddle we can use any shape to fit the text detection. 
So for my ANPR where plates aren't always horizontal...
Is it advisable to use yolo? 

Can you plz answer these one by one. Sorry for troubling you. I'm expecting a non technical answer which is understandable for noobs like me. 


Meanwhile I will try doing text detection to see if texts are getting detected.
Also sorry I don't understand that code you shared. 🥲

**Reply by UserWangZz** [reply]
1. Yes, the recognition needs to ensure that the image area is as small as possible and the recognition text is as complete as possible. Cannot be multi-line text.
2. If the image you crop can meet the above requirements, you can choose not to use it.
3. Millions of training data, not public.
4.More than 5000 images.
5. I can't provide advice on this point. You just need to make sure that the cropped image is a single line of text and surround the text as much as possible.
